### DISCUSSION TITLE
Is the docker compatibility API drifting from spec?

### DISCUSSION THREAD

**Discussion by chesedo** [containers/podman#14641] [Q&A] [answered]
This is somewhat related to #14513, but with a concrete example from the code.

I might understand things wrong because I don't completely understand the docker compatibility API - in which case please correct me. However it seems that the following is all that is needed to start the compat API:
```bash
podman system service -t 5000 &
```

Now from the [docker API](https://docs.docker.com/engine/api/v1.40/#operation/ContainerInspect) it seems container inspect should return a `status.status` string with one of the following:
- created
- running
- paused
- restarting
- removing
- exited
- dead

However, in this libpod code in seems status can also return `initialized`
https://github.com/containers/podman/blob/73713062806aa4c2db25dc62e2fff47406085dc8/libpod/define/containerstate.go#L43-L71

Am I looking at the correct docker compatibility API code? Should it be drifting from the docker spec?

**Comment by Luap99** [comment, marked as answer]
These are just the internal libpod names that are exposed via the podman cli. The docker compat api should always return correct docker output, if not file a bug with a reproducer.
This is the compat api code:
https://github.com/containers/podman/blob/c936d1e61154b6826e9d8df46e9660aba6c86cfe/pkg/api/handlers/compat/containers.go#L277

**Comment by chesedo** [comment]
Thanks, I was looking at the wrong file :grin:

This is a bit hard to reproduce as we don't see it often - the comment in libpod also states "we don't expect to see it [initialized] much" :sweat_smile: 

But I tried to explain my logic in PR #14700

**Reply by mheon** [reply]
You can manually get an Initialized container by running `podman init` on a container in the Created state
